### DISCUSSION TITLE
randomness on devnet

### DISCUSSION THREAD

**Discussion by damiennarozniak** [aptos-labs/aptos-developer-discussions#142] [Questions] [answered]
Hi,

does the next devnet release include
aptos_framework::randomness ?

thanks,

**Comment by 0xmaayan** [comment, marked as answer]
Randomness in currently on devnet, can see the module is under 0x1 account
https://explorer.aptoslabs.com/account/0x0000000000000000000000000000000000000000000000000000000000000001/modules/code/randomness?network=devnet

**Reply by damiennarozniak** [reply]
Hi,
Thanks for answer, i get an error submitting tx on devnet "Randomness APIs calls must originate from a private entry function with...", this is taking the lottery_secure in the core repo. Any reason? thanks

**Reply by 0xmaayan** [reply]
Could you share a code snippet of what you are trying to do?

**Reply by damiennarozniak** [reply]
Hi,

just replicated  https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-41.md raffle's example,
I get this error calling "random_pick_winner" from explorer! 
thanks for hint
